webxrlayers-1.bs

```diff
@@ -0,0 +1,730 @@
+    application to scale down the main eye buffer resolution on low performance systems, but keeping essential information, such as text or a map, in its own layer at a higher
+    resolution.
+
+ * <b>Legibility/visual fidelity</b> The resolution for eye-buffers for 3D world rendering can be set to relatively low values especially on low performance systems. It would
+    be impossible to render high fidelity content, such as text, in this case. Each layer may have its own resolution and it will be re-sampled only once by the compositor (in contrary
+    to the traditional approach with rendering layers via WebGL where the layer's content got re-sampled at least twice: once when rendering into WebGL
+    eye-buffer (and losing a lot of details due to limited eye-buffer resolution) and the second time by the compositor).
+
+ * <b>Power consumption / battery life</b> Due to reduced rendering pipeline, the lack of double sampling and no need to update the layer's rendering each frame, the power
+    consumption is expected to be improved.
+
+ * <b>Latency</b> Pose sampling for composition layers may occur at the very end of the frame and then certain reprojection techniques could be used to update the layer's pose to
+    match it with the most recent HMD pose. This may significantly reduce the effective latency for the layers' rendering and as a result improve overall experience.
+
+ISSUE: how to do hit testing? Should it be done by the UA?
+
+</section>
+
+
+Terminology {#terminology}
+-----------
+
+Application flow {#applicationflow}
+----------------
+
+<section class="non-normative">
+
+If an author wants to use layers, they have to go through these steps:
+ 1. request support through {{XRPermissionDescriptor/requiredFeatures}} or {{XRPermissionDescriptor/optionalFeatures}} in {{XRSystem/requestSession()}}
+ 1. create a XRWebGLLayerFactory or XRMediaLayerFactory
+ 1. create layers with these factories
+ 1. add the layers to {{XRRenderStateInit}} and call {{XRSession/updateRenderState()}}
+ 1. during {{XRSession/requestAnimationFrame()}} for webgl layers, draw content each gl layer
+
+</section>
+
+Model {#model}
+======
+
+In order for the applications to signal their interest in using layers during a session, 
+the session must be requested with appropriate [=feature descriptor=]. The string <dfn>layers</dfn> is introduced 
+by this module as a new valid feature descriptor for WebXR Layers feature.
+
+<div class="example">
+The following code requests layers as an optional feature.
+
+<pre highlight="js">
+navigator.xr.requestSession('immersive-vr', {
+    optionalFeatures: ['layers']
+  }
+</pre>
+</div>
+
+Layers are only supported for XRSessions created with XRSessionMode of {{XRSessionMode/"immersive-vr"}} 
+or {{XRSessionMode/"immersive-ar"}}. {{XRSessionMode/"inline"}} sessions must not support layers.
```

Comment:
```suggestion
or {{XRSessionMode/"immersive-ar"}}. {{XRSessionMode/"inline"}} sessions MUST not support layers.
```

If this is intended to be a normative requirement (which it sounds like it is) then the all-caps 'MUST' should be used to indicate that.

Reply:
done